Sources/Sentry/SentryScope.m

```diff
@@ -255,78 +262,80 @@ - (void)setLevel:(enum SentryLevel)level {
 }
 
 - (SentryEvent * __nullable)applyToEvent:(SentryEvent *)event maxBreadcrumb:(NSUInteger)maxBreadcrumbs {
-    if (nil != self.tagDictionary) {
-        if (nil == event.tags) {
-            event.tags = self.tagDictionary;
-        } else {
-            NSMutableDictionary *newTags = [NSMutableDictionary new];
-            [newTags addEntriesFromDictionary:self.tagDictionary];
-            [newTags addEntriesFromDictionary:event.tags];
-            event.tags = newTags;
+    @synchronized (self) {
+        if (nil != self.tagDictionary) {
+            if (nil == event.tags) {
+                event.tags = self.tagDictionary.copy;
+            } else {
+                NSMutableDictionary *newTags = [NSMutableDictionary new];
+                [newTags addEntriesFromDictionary:self.tagDictionary];
+                [newTags addEntriesFromDictionary:event.tags];
+                event.tags = newTags;
+            }
         }
-    }
 
-    if (nil != self.extraDictionary) {
-        if (nil == event.extra) {
-            event.extra = self.extraDictionary;
-        } else {
-            NSMutableDictionary *newExtra = [NSMutableDictionary new];
-            [newExtra addEntriesFromDictionary:self.extraDictionary];
-            [newExtra addEntriesFromDictionary:event.extra];
-            event.extra = newExtra;
+        if (nil != self.extraDictionary) {
+            if (nil == event.extra) {
+                event.extra = self.extraDictionary.copy;
+            } else {
+                NSMutableDictionary *newExtra = [NSMutableDictionary new];
+                [newExtra addEntriesFromDictionary:self.extraDictionary];
+                [newExtra addEntriesFromDictionary:event.extra];
+                event.extra = newExtra;
+            }
         }
-    }
 
-    if (nil != self.userObject) {
-        event.user = self.userObject;
-    }
-    
-    NSString *releaseName = [self releaseString];
-    if (nil != releaseName && nil == event.releaseName) {
-        // release can also be set via options but scope takes precedence.
-        event.releaseName = releaseName;
-    }
-    
-    NSString *dist = self.distString;
-    if (nil != dist && nil == event.dist) {
-        // dist can also be set via options but scope takes precedence.
-        event.dist = dist;
-    }
-    
-    NSString *environment = self.environmentString;
-    if (nil != environment && nil == event.environment) {
-        // environment can also be set via options but scope takes precedence.
-        event.environment = environment;
-    }
+        if (nil != self.userObject) {
+            event.user = self.userObject.copy;
```

Comment:
So this will not work, because we need to implement `NSCoding` for `SentryUser`

Reply:
Will fix it, thanks